include/regs.h

```diff
@@ -166,4 +166,141 @@
+    /*  8 */ HREG_MODE_PRINT_MEMORY, // print memory at a specified address
+    /*  9 */ HREG_MODE_UNUSED_9,
+    /* 10 */ HREG_MODE_PLAY, // various debug controls for the Play state
+    /* 11 */ HREG_MODE_PERSPECTIVE, // set various properties of the view perspective
+    /* 12 */ HREG_MODE_INPUT_TEST, // displays inputs from the specified controller
+    /* 13 */ HREG_MODE_SCHED, // toggle various special vi special features
+    /* 14 */ HREG_MODE_UNUSED_14,
+    /* 15 */ HREG_MODE_PRERENDER,
+    /* 16 */ HREG_MODE_SETUP_FRAME, // set the base screen color (and by extension, shrink window color too)
+    /* 17 */ HREG_MODE_SCENE_CONFIG, // toggle drawing the scene config
+    /* 18 */ HREG_MODE_UNUSED_18,
+    /* 19 */ HREG_MODE_LETTERBOX, // toggle logging and view values for the letterbox system
+    /* 20 */ HREG_MODE_HEAP_FREE_BLOCK_TEST, // enable or disable heap free block test
+    /* 21 */ HREG_MODE_VI
+} HRegMode;
+
+// HREG_MODE_UCODE_DISAS
+#define R_UCODE_DISAS_TOGGLE      HREG(81) // < 0 toggles and prints some hardware reg info, > 0 only toggles disas
+#define R_UCODE_DISAS_LOG_MODE    HREG(82) // 1 and 2 print counts, 3 enables fault client, 4 disables open/close disps
+#define R_UCODE_DISAS_LOG_LEVEL   HREG(83) // enables various logging within the dissasembler itself
+#define R_UCODE_DISAS_TOTAL_COUNT HREG(84) // only displays, changing this reg will not set its value
+#define R_UCODE_DISAS_VTX_COUNT   HREG(85) // only displays, changing this reg will not set its value
+#define R_UCODE_DISAS_SPVTX_COUNT HREG(86) // only displays, changing this reg will not set its value
+#define R_UCODE_DISAS_TRI1_COUNT  HREG(87) // only displays, changing this reg will not set its value
+#define R_UCODE_DISAS_TRI2_COUNT  HREG(88) // only displays, changing this reg will not set its value
+#define R_UCODE_DISAS_QUAD_COUNT  HREG(89) // only displays, changing this reg will not set its value
+#define R_UCODE_DISAS_LINE_COUNT  HREG(90) // only displays, changing this reg will not set its value
+#define R_UCODE_DISAS_SYNC_ERROR  HREG(91) // only displays, changing this reg will not set its value
+#define R_UCODE_DISAS_LOAD_COUNT  HREG(92) // only displays, changing this reg will not set its value
+#define R_UCODE_DISAS_DL_COUNT    HREG(93) // only displays, changing this reg will not set its value
+
+// HREG_MODE_PRINT_MEMORY
+#define R_PRINT_MEMORY_INIT     HREG(94) // set to `HREG_MODE_PRINT_MEMORY` when init is complete
+#define R_PRINT_MEMORY_TRIGGER  HREG(81) // set to a negative number to print memory
+#define R_PRINT_MEMORY_ADDR     HREG(82) // upper 16 bits of the address to print from
+#define R_PRINT_MEMORY_SIZE     HREG(83) // value * 0x10 = length of the log
+
+// HREG_MODE_PLAY
+#define R_PLAY_INIT              HREG(94) // set to `HREG_MODE_PLAY` when init is complete
+#define R_RUN_UPDATE             HREG(81)
+#define R_RUN_DRAW               HREG(82)
```

Comment:
Seems like these need a system prefix of some kind (maybe based on the mode?), otherwise names like these are very ambiguous. I wouldn't be surprised if these were originally in the file itself given that the vast majority of each mode's associated regs are only used in a single file.

Reply:
oh, yea i just forgot to put a prefix on PLAY. 
regarding your second point, possibly, but ironically PLAY is one of the few that have overlap with other files lol.
I also think it helps to understand the system to have them all in one place, even if that means its own header for hayakwa debug specifically

Reply:
I'd agree that it has somewhat outgrown how regs.h is usually used at this point.

Reply:
sort has, and sort of not. its still just giving intelligible names to the regs, with the added extra complexity of the modes that it can switch between

Reply:
Well, that means now it's a system, not just a set of numbers, right? If there was another similar system (which thankfully I don't think there is), it would be confusing to keep it all in the same header.

Reply:
I think either way there is value in keeping the "aliases" for regs all in one place, so you know where they are without question. but yea, can see what others think

Reply:
Yeah, the main issue here is that there's multiple aliases depending on context, which AFAIK is new.